.github/workflows/task-unit-test.yml

```diff
@@ -25,11 +25,10 @@ jobs:
     container: ${{ inputs.container || null }}
     defaults:
       run:
-        shell: bash -l -eo pipefail {0}
+        # Use bash as default shell for every OS other than alpine
+        shell: ${{ inputs.container && contains(inputs.container, 'alpine') && 'sh -l -eo pipefail {0}' || 'bash -l -eo pipefail {0}' }}
```

Comment:
Ugly, but couldn't make it work otherwise (Alpine doesn't have bash by default, so it fails immdietly, other os cannot accept the `sh -l -eo pipefail {0} command since their default shell do not support -o pipefail)

Reply:
you can use the steps we have on RediSearch, we handles it in 2 steps

Reply:
I see. Frankly, I don't think that one of the approaches is better than the other (both a bit ugly...)